### ISSUE TITLE
Infrastructure: Upgrade SteamVR

### ISSUE THREAD

**Issue by bastianilso** [med-material/Whack_A_Mole_VR#197]
As part of resolving issue #183 we need to upgrade our version of SteamVR. This process involves several steps.

- We need to see what version of SteamVR we are currently using.
- We need to identify which version of SteamVR is the next to upgrade to, given that we are still on Unity 2019.x. (See all releases here: https://github.com/ValveSoftware/steamvr_unity_plugin/releases)
- There will most likely be API changes that we need to follow, in order to make Whack-A-Mole VR work again - related to controller bindings.

First step in this issue is to research it and see what is possible. @lucasmnt please let me know what you might find. I have assigned you to this task, but if @Xav1204 would rather want to work on it, feel free to swop. 🙂

**Comment by bastianilso** [comment]
@lucasmnt @Xav1204  if one of you have time this week, I would like you to start a branch where you investigate this. Please inform in this issue what discoveries you make about it.

**Comment by Xav1204** [comment]
![image](https://user-images.githubusercontent.com/73820233/194837917-fc869f6d-40bd-495c-ace4-4dabe310babb.png)

@bastianilso Is it normal that we don't have steamVR plugin in the package manager?

**Comment by Xav1204** [comment]
@bastianilso It's good, I found the version in the following file : "\Assets\SteamVR\readme.txt".

We are actually in the version 2.0.1

![image](https://user-images.githubusercontent.com/73820233/194840954-a5575c5d-b172-4753-a00b-73d870d204a0.png)

**Comment by bastianilso** [comment]
@Xav1204 older versions of SteamVR are not provided through the Unity Package Manager. 
OK, so now that we know which version we are at, next we need to know which version we can upgrade to, while staying on Unity 2019.x. 

You can download find unity packages of all steam vr versions from the link I provided:
https://github.com/ValveSoftware/steamvr_unity_plugin/releases

Hope that helps.

**Comment by Xav1204** [comment]
![image](https://user-images.githubusercontent.com/73820233/194853176-7f9f58b5-82ef-4cac-9ba6-92ffaa6c819f.png)
> For the second question, we can upgrade our steamVR version until the lastest one that is the 2.7.1. In the documentation, it's write that a package will be installed. Therefore, as the package that we need is installed so we won't encounter a problem with this. However, I don't know what will happen with Unity XR as we are under v2020. 

![image](https://user-images.githubusercontent.com/73820233/194862611-47eb4a55-900e-4261-9a52-27275a3d3569.png)
> For the third question, yes we will have some changes to make in order to make that the game work again. 


I will try on a new branch to make this changes and I will tell you if I encounter any problems during the process.

**Comment by Xav1204** [comment]
@bastianilso I succeeded to update steamVR until its latest version. I just need to try with the VR headset if the controllers are still working.

**Comment by bastianilso** [comment]
Todo for the SteamVR:

1. Document the errors you get and how you solved them.
2. If the above does not work as expected, try creating a new empty project with Unity 2019.3.13f1 and import the SteamVR unity package. Confirm whether it works without Whack-A-Mole VR components first. Try to replicate one-by-one the CameraRig used in Whack-A-MoleVR. You can try to import the default CameraRig which we know works and go from there.

**Comment by Xav1204** [comment]
@bastianilso it's better now, we see the controller, the cursor input and movement are correct and we don't have problem with the camera direction. I just need to investigate why the hand and the blue circle don't appear.
![image](https://user-images.githubusercontent.com/73820233/197719076-8ccc5404-5a5d-40b1-83b1-f9312163c6bc.png)

**Comment by Xav1204** [comment]
@bastianilso you will find attached the report about the steamVR upgrade.

[steamVR.pdf](https://github.com/med-material/Whack_A_Mole_VR/files/9937918/steamVR.pdf)

**Comment by Xav1204** [comment]
@bastianilso yes, upgrade SteamVr is the next step to do in the different upgrades that we need to do. As soon as I will finish the prism effect, I will do this issue.

**Comment by Xav1204** [comment]
@bastianilso I've found the problem with the hand/controller position and rendering. I will write you a report to explain what I've done to solve the problem. We can so upgrade SteamVR to the latest version.

**Comment by Xav1204** [comment]
@bastianilso Here the report about the problem that we had with the hand position and rendering : 
[steamVR.pdf](https://github.com/med-material/Whack_A_Mole_VR/files/10345045/steamVR.pdf)
